[MRG] Added CV return predictions for predict/predict_proba

okz12 opened the pull request:
Fix for #13478

This allows the user to extract predictions or prediction probabilities using 'return_predictions' argument for cross_validate. The argument takes 'predict_proba' or 'predict' as a string to return the predictions.

The approach has been to have `_fit_and_score` run inference on the cross-validation split and return it with the indices. Once all the indices and predictions have been returned to the `cross_validate` function, they are ordered according to the dataset and returned.

```
from sklearn.linear_model import LogisticRegression
from sklearn.model_selection import cross_validate
from sklearn.datasets._samples_generator import make_classification

n_samples = 25
n_classes = 2
clf = LogisticRegression(random_state=0)
X, y = make_classification(n_samples=n_samples, n_classes=n_classes)

ret = cross_validate(clf, X, y, return_train_score=False,
                     return_estimator=True, return_predictions='predict')
ret['predictions']
>>> array([0, 1, 1, 0, 0, 1, 0, 1, 0, 1, 1, 0, 0, 0, 1, 0, 0, 1, 0, 1, 0, 1,
       0, 1, 1])
```

lucyleeow commented:
Thanks @okz12, I will follow this one. And thank you but, cross_val_predict doesn't work - I need the scores and the predictions and I would like the cv indicies but this one is not essential.

okz12 commented:
@jnothman 
The PR is ready for review (carrying on from #15413, sorry for the separate PR). Also, I'm not sure why the 2 checks are failing.

thomasjpfan committed:
CLN Minor adjustment for potential overflow in hist gradient boosting (#17410)

NicolasHug committed:
DOC minor update to classification_report docstring (#16899)

Co-authored-by: Guillaume Lemaitre <g.lemaitre58@gmail.com>
Co-authored-by: Andreas Mueller <t3kcit@gmail.com>

thomasjpfan committed:
ENH Makes _deprecate_positional_args slightly faster (#17246)

alfaro96 committed:
DOC Specify shape with parentheses in decision tree guide (#17873)

thomasjpfan committed:
CLN Uses bool instead of np.bool (#17875)

jeremiedbb committed:
MNT Centralize all param checks in KMeans (#17861)

strivn committed:
DOC Fix documentation of default values in tree exports (#17834)

Co-authored-by: Thomas J. Fan <thomasjpfan@gmail.com>
Co-authored-by: Juan Carlos Alfaro Jiménez <JuanCarlos.Alfaro@uclm.es>

yashika51 committed:
DOC Added missing attributes to OneVsRest Classifier (#17844)

Co-authored-by: Thomas J. Fan <thomasjpfan@gmail.com>

franslarsson committed:
DOC Fix default values in multioutput.py docstrings (#17841)

Co-authored-by: Juan Carlos Alfaro Jiménez <JuanCarlos.Alfaro@uclm.es>
Co-authored-by: Guillaume Lemaitre <g.lemaitre58@gmail.com>

jeremiedbb committed:
FIX MeanShift when estimated bandwidth is 0 (#17742)

shail2512-lm10 committed:
DOC added attributes in LassoLarsCV (#17838)

glemaitre committed:
EXA Improve the alphas values in MLP example (#17880)

Co-authored-by: 叶贞明 TaiChiTiger <sakyayj@126.com>

raybellwaves committed:
DOC add Gitter link to readme (#17883)

jeremiedbb committed:
CI Fix pre-build checks to handle compilers from build_ext options (#16193)

jeremiedbb committed:
CI import setuptools monkeypatch distutils commands (#17887)

yashika51 committed:
ENH Added parameter clip to MinMaxScaler (#17833)

alfaro96 committed:
DOC Specify shape with parentheses in ensemble user guide (#17888)

marenwestermann committed:
add dual_gap_ attribute (#17882)

alfaro96 committed:
DOC Add Twitter link to README.rst (#17891)

biprateep committed:
removed os import (#17893)

biprateep committed:
MNT removed repeated definition of ranked_exposure (#17895)

willpeppo committed:
EXA Avoid redundant numpy imports in plot_release_highlights.py (#17903)

mikeaalv committed:
ENH remove default y value in sklearn.kernel_ridge.KernelRidge.fit (#17906)

biprateep committed:
MNT Fixed import statements (#17901)

* fixed import statements

* fixed linting issues

* keep numpy import within setup_module

Carreau committed:
DOC Fix numpydoc section name Return -> Returns (#17879)

From a cursory glance this is the only place I found in sklearn (and to a wider extend scipy/numpy) that uses Return instead of Returns as described in the numpydoc format.

mikeaalv committed:
[MRG] update the documents (default value) in the function sklearn.kernel_ridge.KernelRidge (#17904)

* update the documents (default value) in the function kernel_ridges.py

* update the array size notation

* update sample_weight array size

Hoda1394 committed:
DOC updated documentation default value for gradient_boosting (#17909)

joshuacwnewton committed:
[MRG] ENH Add parameter to sklearn.manifold.TSNE to transition to squaring non-Euclidean distance metrics by default (#12401) (#17662)

jeremiedbb committed:
DOC document and deprecate missing attributes in MiniBatchKMeans (#17864)

Probinette4 committed:
DOC update to text/CountVectorizer token_pattern parameter default value (#17898)

* Updated documenation in CountVectorizer function for the token_parameter default value

* Removed the r before the string, error in flake8

* Additional change to see if flake8 error solved

Co-authored-by: Preston <pkrobinet@presby.edu>

franslarsson committed:
DOC Fix documentation of default values in sklearn.utils.optimize.py (#17881)

* update docstring according to guideline

* add space for consistency

franslarsson committed:
DOC Fix documentation of default values in sklearn.utils._encode.py (#17859)

thomasjpfan committed:
CI Build with cython prerelease (#17908)

willpeppo committed:
MNT code hygiene on _mocking.py (#17894)

glemaitre committed:
FIX remove pytest warning in dataset/tests/test_common.py (#17828)

Co-authored-by: Thomas J. Fan <thomasjpfan@gmail.com>

lucyleeow committed:
DOC Clarify return of cross_val_predict (#17851)

chaitanyamogal committed:
DOC fix plot position in statistical_inference/settings.rst (#17919)

changed position of the image

thomasjpfan committed:
DOC Small doc formatting fixes in decomposition (#17918)

TahiriNadia committed:
DOC cleaning parameter docstrings in decomposition/_dict_learning.py (#17896)

* update formatting for default value for consistency'

* Update sklearn/decomposition/_dict_learning.py

Co-authored-by: Nicolas Hug <contact@nicolas-hug.com>

* update formatting for alpha default value for consistency

* Update sklearn/decomposition/_dict_learning.py

Co-authored-by: Nicolas Hug <contact@nicolas-hug.com>

Co-authored-by: Nicolas Hug <contact@nicolas-hug.com>

thomasjpfan committed:
CI Temporary fix for conda-forge:: compilers on osx (#17913)

* CI Temporary fix for conda-forge on osx

* DOC Update doc on compiler version

* CLN Apply suggestions

david-cortes committed:
DOC Missing input_type to FeatureHasher (#17922)

m-sche committed:
DOC Fix error text when null_space fails with ARPACK (#17927)

alfaro96 committed:
MNT Avoid duplicate validation of data in supervised NN (#17849)

alfaro96 committed:
MNT Use substitutions for minimum dependencies in documentation (#17931)

thomasjpfan committed:
BUG Do not overwrite X when copy_X=True and Gram='auto' (#17914)

alexhenrie committed:
PERF Get rid of intermediate array in binary_log_loss (#17932)

chaitanyamogal committed:
EXA change in position of plot image (#17924)

oj-lappi committed:
DOC Fix math notation error in description of CCP (#16450)

Breiman, Friedman, Ohlsen, Stone use a T with a tilde to denote the terminal nodes of a tree. A simple T would imply we are counting the nodes in the whole tree.

Commit:
TST Add test for multinomial identifiability constraint in logistic regression (#16791)

* MNT import from numpy.testing in logistic regression

* TST add test for symmetric identifiability constraint

* address review comments

* address review comments again

* reduce atol

alfaro96 committed:
DOC Specify base_estimator=None in CalibratedClassifierCV (#17393)

thomasjpfan committed:
ENH Uses binned values from training to find missing values (#17736)

* ENH Uses training data to find missing values

* CLN Uses the binned data to find missing value

* ENH Uses binned values from training to find missing values

thomasjpfan committed:
ENH Adds support for multimetric callable return a dictionary (#15126)

marenwestermann committed:
DOC sparse_coef_ documentation for MultiTaskElasticNet and MultiTaskLasso (#17884)

jeremiedbb committed:
MNT Refactor center initialization in KMeans (#17928)

* refactor center init in KMeans

* address comments

jeremiedbb committed:
TST More cleaning/refactoring in KMeans tests (#17885)

ForrestCKoch committed:
FIX Sets dtype for pairwise_distances when metric='seuclidean' (#15730)

Co-authored-by: Jérémie du Boisberranger <34657725+jeremiedbb@users.noreply.github.com>
Co-authored-by: Chiara Marmo <cmarmo@users.noreply.github.com>
Co-authored-by: Thomas J. Fan <thomasjpfan@gmail.com>

SylvainLan committed:
FIX Add warning for GP's hyperparameters (#12673)

* Add a warning for GP Regressor if optimized hyperparameter matches one of its bounds

* pep8 fix

* Add a test and use np.isclose

* More verbose warning message

* remove print

* Ad a more precise warning and update tests

* fix warning and update tests

* fix test dims

* fix multiple dimensions test

* fix pep8

* change variable name

* add warning for GP classifier

* tests for gpc

* pep8 fix

* put the warning in a function

* add comment to warning function

* update name in gpr and gpc

* update whats_new

* update whats_new

* update whats_new

* remove merge conflict text

* add WhiteKernel

* put the log in the last version

alfaro96 committed:
FIX Change MRO for some estimators (#17934)

noelano committed:
PERF Call KDTree in mutual_info to reduce memory footprint (#17878)

Co-authored-by: Thomas J. Fan <thomasjpfan@gmail.com>

thomasjpfan committed:
CI Remove unneeded check for python 3.5 (#17948)

marenwestermann committed:
FIX inconsistent hyperparameter and bounds issue in GP kernels (#17946)

thomasjpfan committed:
TST Test for both solvers in test_nmf_sparse_input (#17947)

alfaro96 committed:
DOC Fix typo in the KNeighborsClassifier documentation (#17958)

alfaro96 committed:
MNT Remove unnecessary try and except for numpy 1.3 (#17957)

jeremiedbb committed:
MNT import setuptools before distutils + add some comments (#17956)

marenwestermann committed:
DOC add attribute active_ to LarsCV (#17886)

thomasjpfan committed:
TST Moving testing code in quad_tree to test_quad_tree (#17968)

lucyleeow committed:
DOC Expand on sigmoid and isotonic in calibration.rst (#17725)

rubywerman committed:
DOC Change "Other Versions" to "Download Docs" in dropdown (#17792)

franslarsson committed:
DOC Fix documentation of defaults in sklearn.utils.class_weight (#17981)

bmaisonn committed:
DOC Fix documentation of default values in multiclass.py (#17986)

bmaisonn committed:
cleanup optional keywords in the documentation (#17990)

bmaisonn committed:
DOC Fix documentation of default values in validation.py (#17988)

amy12xx committed:
TST Added test for Incremental pca value error fix (#17985)


Co-authored-by: Allan <allanbutler9@gmail.com>
Co-authored-by: Roman Yurchak <rth.yurchak@gmail.com>

saket404 committed:
DOC fix dataset loading documentation (#17993)

amy12xx committed:
FIX AffinityPropagation fix results with float32 input (#17995)

FrancescoCasalegno committed:
Distances for agglomerativeclustering (#17984)

Co-authored-by: Michael Riedmann <michael_riedmann@live.com>
Co-authored-by: Emilie Delattre <emilie.delattre@epfl.ch>
Co-authored-by: EmilieDel <47669575+EmilieDel@users.noreply.github.com>

RobRomijnders committed:
Fix typo in docstring for GaussianMixture. (#17998)

jjerphan committed:
[MRG] Use joblib.Parallel for CalibratedClassifierCV.fit (#17107)

Co-authored-by: Thomas J. Fan <thomasjpfan@gmail.com>

guillaume-chevalier committed:
DOC Add Neuraxle to Related Projects page (#16100)

Co-authored-by: Roman Yurchak <rth.yurchak@gmail.com>

thomasjpfan committed:
TST Default skip network tests (#18002)

mani2106 committed:
MNT Remove redundant typing import (#18004)

thomasjpfan committed:
FIX Fixes CountVectorizer sample invariance with max_features (#18016)

jeremiedbb committed:
[MRG] Fix KMeans convergence when tol==0 (#17959)

franslarsson committed:
DOC Fix doc of defaults in sklearn.utils.multiclass.py (#18021)

napsternxg committed:
Fix #16998 classification_report with output_dict=True should not raise AttributeError  (#17777)

thomasjpfan committed:
CI Fixes circleci doc building by adding path to conda in bash script (#18022)

jeremiedbb committed:
MNT Add asv benchmark suite (#17026)

thomasjpfan committed:
[MRG] ENH Does not overwrite ylabel in partial dependence if present (#15776)

zhaofeng-shu33 committed:
DOC fix typo in plot_tweedie_regression_insurance_claims.py (#18026)

glemaitre committed:
DOC make warning more obvious in example regarding feature importance (#18024)

lucyleeow committed:
DOC Fix return docstring cross_val_predict (#18035)

JinLi711 committed:
DOC Changed Doc to replace "optional" with "default" (#17982)

Co-authored-by: Thomas J. Fan <thomasjpfan@gmail.com>
Co-authored-by: Roman Yurchak <rth.yurchak@gmail.com>

dachosen1 committed:
ENH warn in clustering metrics given continuous input (#17911)

ogrisel committed:
DOC add section on debugging ARM64 problems (#18033)

franslarsson committed:
DOC Fix doc of defaults in sklearn.utils.estimator_checks.py (#18037)

franslarsson committed:
DOC Fix doc of defaults in sklearn.calibration.py (#18034)

lucyleeow committed:
DOC Add doc for permutation_test_score (#17373)

rth committed:
Arm64 CI setup with TravisCI (#17996)

lucyleeow committed:
DOC Add example of using CV iterator to split train test (#17550)

lucyleeow committed:
DOC Update plot_permutation_test_for_classification.py (#17385)

tijanajovanovic committed:
DOC cleaning parameter docstrings in feature_extraction/image.py (#17900)

Co-authored-by: gagileo <babic.dr@gmail.com>
Co-authored-by: Guillaume Lemaitre <g.lemaitre58@gmail.com>

ckastner committed:
TST: Use assert_allclose instead of equality in FP comparison (#18053)

lucyleeow committed:
DOC Use math notation in log_loss metric (#18049)

franslarsson committed:
DOC update docstring default values in KernelDensity to follow doc guideline (#18050)

cmarmo committed:
FEA Add iterable values support for dictvectorizer (#17367)



Co-authored-by: Peng Yu <peng.yu@shopify.com>
Co-authored-by: Thomas J. Fan <thomasjpfan@gmail.com>
Co-authored-by: Joel Nothman <joel.nothman@gmail.com>

franslarsson committed:
DOC Fix doc of defaults in sklearn.utils.validation.py (#18048)

Co-authored-by: Guillaume Lemaitre <g.lemaitre58@gmail.com>

franslarsson committed:
DOC Fix doc of defaults in sklearn.metrics._scorer.py (#18058)

thomasjpfan committed:
DOC Improves sidebar navigation for user guide (#16964)

justmarkham committed:
DOC Minor documentation fixes (#18056)

franslarsson committed:
DOC Fix doc of defaults in sklearn.utils.graph.py (#18064)

franslarsson committed:
DOC Fix doc of defaults in manifold._spectral_embedding.py (#18061)

franslarsson committed:
DOC Fix doc of defaults in sklearn.manifold._mds.py (#18060)

glemaitre committed:
Update whats new formatting

ogrisel committed:
MAINT mark AgglomerativeClustering's new compute_distances as |Enhancement|

glemaitre committed:
DOC update the whats new (#18070)

thomasjpfan committed:
DOC Adds whats new to 0.23.2 for bug fix (#17912)

Co-authored-by: Joel Nothman <joel.nothman@gmail.com>
Co-authored-by: Guillaume Lemaitre <g.lemaitre58@gmail.com>

jeremiedbb committed:
TST More cleaning / refactoring in KMeans tests part 2 (#17973)

* remove test_minibatch_set_init_size already included in test_minibatch_init_size

* remove test_minibatch_tol test nothing

* improve and parametrize test_predict_dense_sparse

* remove test_full_vs_elkan already in test_predict and test_elkan_results

* simplify test_n_init

* fix test_centers_not_mutated

* cleaner test_kmeans_init_fitted_centers

* simpler test_kmeans_iter_attribute

* parametrize test_kmeans_empty_cluster_relocated dense sparse

* cln test_warning_elkan_1_cluster

* single double quote uniformisation

* add comment for test_euclidean_distance

* parametrize test_sample_weight_unchanged

* join test_predict, test_fit_predict and test_predict_equal_labels

* remove test_sparse_mb_k_means_callable_init & test_mini_batch_k_means_random_init_partial_fit already included it test_all_init

* Estimator -> algorithm

ogrisel committed:
MAINT scipy-dev failures because of too numerically strict tests (#18071)

solegalli committed:
DOC Add Feature engine to Related Projects page (#18012)

Co-authored-by: Soledad Galli <solegalli1@gmail.com>

glemaitre committed:
MNT update date and links on the index webpage (#18081)

sadakmed committed:
DOC Fix doc of defaults in gaussion_process._gpr.py (#18075)

ckastner committed:
TST Increase absolute tolerance for RANSACRegressor test (#18082)

An absolute difference of 1.45081487e-13 has been observed on 64-bit PowerPC.

franslarsson committed:
DOC Fix doc of defaults in sklearn.metrics._ranking.py (#18093)

FelixWick committed:
FEA allow unknowns in OrdinalEncoder transform (#17406)

jnothman committed:
DOC Some what's new tweaks

NicolasHug committed:
MNT remove pre-dispatch and verbose from CalibratedClassifierCV (#18030)

bmaisonn committed:
DOC Fix documentation of default values in fastica.py (#17989)

bmaisonn committed:
Fix MemoryError exception when calling barycenter_weights with large inputs (#17997)

xun-tang committed:
DOC Address numpydoc validation to GradientBoosting* (#15477)

Co-Authored-By: Roman Yurchak <rth.yurchak@gmail.com>
Co-authored-by: Matteo Visconti dOC <mvdoc@berkeley.edu>
Co-authored-by: Roman Yurchak <rth.yurchak@gmail.com>
Co-authored-by: Thomas J. Fan <thomasjpfan@gmail.com>

hiteshsom committed:
DOC Update tree.rst about lack of support of categorical data (#18106)

franslarsson committed:
DOC Fix doc of defaults in sklearn.linear_model._theil_sen (#18099)

franslarsson committed:
DOC Fix doc of defaults in sklearn.linear_model._sag.py (#18098)

NicolasHug committed:
[MRG] Prototype 4 for strict check_estimator mode (#17361)

franslarsson committed:
DOC Fix doc of defaults in sklearn.linear_model._omp.py (#18100)

Co-authored-by: Roman Yurchak <rth.yurchak@gmail.com>

thomasjpfan committed:
TST Sets random_state in test_rfe_wrapped_estimator (#18097)

cmarmo committed:
FEA Add Nonnegative LinearRegression (#17578)

Co-authored-by: Nicolas Hug <contact@nicolas-hug.com>
Co-authored-by: Joseph Knox <jknox13@uw.edu>
Co-authored-by: Joseph Knox <joseph.edward.knox@gmail.com>

wmvanvliet committed:
ENH Add alpha_per_target parameter to RidgeCV (#6624)

Co-authored-by: Thomas J. Fan <thomasjpfan@gmail.com>

cmarmo committed:
DOC Fix mpl in plot_partial_dependence_visualization_api (#18109)

bharatr21 committed:
ENH: Add regularization to the main NMF class (#17414)

* ENH: Add regularization to the main NMF class

* Update _nmf with suggestions from code review

Update `_nmf.py` with suggestion from review

Co-authored-by: Adrin Jalali <adrin.jalali@gmail.com>

* Refactor tests, fix linter errors

* Change default value to None

* Revert back to default value of "both"

* Update default value documentation acc to @thomasjpfan

Co-authored-by: Thomas J. Fan <thomasjpfan@gmail.com>

* CLN Places regularization at the end

* Add whatsnew entry

* DOC Fix

* DOC Fix

Co-authored-by: Adrin Jalali <adrin.jalali@gmail.com>
Co-authored-by: Thomas J. Fan <thomasjpfan@gmail.com>

ckastner committed:
TST Change expected result type np.int64 -> np.int (#18089)

Co-authored-by: Thomas J. Fan <thomasjpfan@gmail.com>

timokau committed:
TST Relax init parameter checks for unmutable objects (#17936)

KartikChugh committed:
EXA Improves contour boundaries for K-Nearest Neighbors (#17580)

bharatr21 committed:
DOC Correction in empirical covariance (#17587)

thomasjpfan committed:
ENH Adds support for list of bools in column transformer (#17616)

kaylani2 committed:
DOC Mention the conditions for a ValueError in VarianceThreshold  (#17380)

NicolasHug committed:
DOC minor typo fix (#18119)

tbenjamin-pl committed:
DOC Fixes hard coded title in gmm selection example plot (#17418)

Co-authored-by: Thomas S. Benjamin <tbenjamin@vencorelabs.com>
Co-authored-by: Roman Yurchak <rth.yurchak@gmail.com>

hcars committed:
DOC Add note on the max number of extmath.cartesian arguments (#16406)

Co-authored-by: Henry <hlc5v@virginia.edu>
Co-authored-by: Roman Yurchak <rth.yurchak@gmail.com>

sstalley committed:
ENH Add verbose option to SpectralClustering (#18052)

Co-authored-by: Thomas J. Fan <thomasjpfan@gmail.com>

cmarmo committed:
DOC Mark add_toctree_function safe for parallel read and write (#18123)

martina-zxy committed:
DOC Fix doc of default values in sklearn.neural_network._multilayer_perceptron.py (#18133)

* DOC Fix doc of default values in sklearn.neural_network._multilayer_perceptron.py

* Update sklearn/neural_network/_multilayer_perceptron.py

Co-authored-by: Thomas J. Fan <thomasjpfan@gmail.com>

Co-authored-by: Thomas J. Fan <thomasjpfan@gmail.com>

franslarsson committed:
DOC Update TSNE docstring (#18120)

GaelVaroquaux committed:
DOC on issue triaging process (#17907)

Co-authored-by: Nicolas Hug <contact@nicolas-hug.com>

NicolasHug committed:
DOC Add example comparing PCR and PLS (#17151)

qzhang90 committed:
FIX Fix a cross-platform endian issue in trees (#17644)

* fixed a cross-platform endian issue

* removed a duplicated dtype checking

* Trigger [arm64] CI

* added an entry to doc/whats_new/v0.24.rst

* moved my fix entry to sklearn.tree section

* Update sklearn/tree/_tree.pyx

Added comments

Co-authored-by: Roman Yurchak <rth.yurchak@gmail.com>

* code simplified

* fixed a typo in sklearn/tree/_tree.pyx

* Update doc/whats_new/v0.24.rst

Co-authored-by: Roman Yurchak <rth.yurchak@gmail.com>

* updated doc/whats_new/v0.24.rst

* Update sklearn/tree/_tree.pyx

Added a space after "if", to be PEP8 compliant.

Co-authored-by: Roman Yurchak <rth.yurchak@gmail.com>

* Update sklearn/tree/_tree.pyx

Co-authored-by: Qi Zhang <q.zhang@ibm.com>
Co-authored-by: Olivier Grisel <olivier.grisel@ensta.org>
Co-authored-by: Roman Yurchak <rth.yurchak@gmail.com>

smarie committed:
FIX use specific threshold to discard eigenvalues with 32 bits fp(#18149)

Co-authored-by: Sylvain MARIE <sylvain.marie@se.com>

cmarmo committed:
DOC Clarify triaging role. Remove usage question template. (#18142)

Co-authored-by: Nicolas Hug <contact@nicolas-hug.com>

simonamaggio committed:
DEP Deprecate n_classes_ in GradientBoostingRegressor (#17702)

Co-authored-by: Guillaume Lemaitre <g.lemaitre58@gmail.com>

lucyleeow committed:
DOC Expand Brier score, fix docstring (#18051)

* fix brier

* fix brier

* add ref

* add ref for lint

* Add link to PDF for calibration book chapter.

* Fix broken repeated ref by using qualified identifier

* Fix indentation

Co-authored-by: Olivier Grisel <olivier.grisel@ensta.org>

lucyleeow committed:
DOC Add user guide for permutation_test_score (#18055)

* wip

* add user guide

* fix link to ex

* suggestion

* suggestions

thomasjpfan committed:
MNT Support for case insensitive filesystems (#18151)

RoeiKa committed:
ENH Permutation importance support sample weight  (#16906)

thomasjpfan committed:
CI Disable 32bit pytest xdist (#18161)

fujiaxiang committed:
API Change default value of as_frame in fetch_openml to 'auto' (#17610)

rauwuckl committed:
DOC clarify the kernel gradient for GaussianProcesses (#18115)

dmohns committed:
[MRG] Add Detection Error Tradeoff (DET) curve classification metrics (#10591)

* Initial add DET curve to classification metrics

* Add DET to exports

* Fix DET-curve doctest errors

- Sample snippet in  model_evaluation documentation was outdated.

* Clarify wording in DET-curve computation

- Align to the wording of ranking module to make it consistent.
- Add correct describtion of input and outputs.
- Update and fix non-existent links

* Beautify DET curve documentation source

- Limit line length to 80 characters.

* Expand DET curve documentation

- Add an example plot to show difference between ROC and DET curves.
- Expand Usage Note section with background information and properties
of DET curves.

* Update DET-curve documentation

- Fix typos and some grammar improvements.
- Use named references to avoid potential conflicts with other sections.
- Remove unneeded references and improved existing ones by using e.g.
using versioned links.

* Select relevant DET points using slice object

* Remove some dubiety from DET curve doc-string

* Add DET curve contributors

* Add tests for DET curves

* Streamline DET test by using parametrization

* Increase verbosity of DET curve error handling

- Explicitly sanity check input before computing a DET curve.
- Add test for perfect scores.
- Adapt indentation style to match the test module.

* Add reference for DET curves in invariance test

* Add automated invariance checks for DET curves

* Resolve merge artifacts

* Make doctest happy

* Fix whitespaces for doctest

* Revert unintended whitespace changes

* Revert unintended white space changes #2

* Fix typos and grammar

* Fix white space in doc

* Streamline test code

* Remove rebase artifacts

* Fix PR link in doc

* Fix test_ranking

* Fix rebase errors

* Fix import

* Bring back newlines

- Swallowed by copy/paste

* Remove uncited ref link

* Remove matplotlib deprecation warning

* Bring back hidden reference

* Add motivation to DET example

* Fix lint

* Add citation

* Use modern matplotlib API

Co-authored-by: Jeremy Karnowski <jeremy.karnowski@gmail.com>
Co-authored-by: Julien Cornebise <julien@cornebise.com>
Co-authored-by: Daniel Mohns <daniel.mohns@zenguard.org>

thomasjpfan committed:
DOC Fixes whats new by removing undefined module (#18172)

Commit:
DOC improve model evaluation a bit (#18166)

amy12xx committed:
FIX Fixes ClassifierChain issue when a tuple is passed to order (#18124)

* fix error tuple passed to order

* fix linting

* Update sklearn/tests/test_multioutput.py

Co-authored-by: Roman Yurchak <rth.yurchak@gmail.com>

* updated test and whatsnew

* doc fix

* Update doc/whats_new/v0.24.rst

Co-authored-by: Roman Yurchak <rth.yurchak@gmail.com>

* Update sklearn/tests/test_multioutput.py

Co-authored-by: Thomas J. Fan <thomasjpfan@gmail.com>

* Update sklearn/tests/test_multioutput.py

Co-authored-by: Thomas J. Fan <thomasjpfan@gmail.com>

* code review fix

* code review fix

* added test

* code review fix

* Update sklearn/multioutput.py

Co-authored-by: Joel Nothman <joel.nothman@gmail.com>

Co-authored-by: Roman Yurchak <rth.yurchak@gmail.com>
Co-authored-by: Thomas J. Fan <thomasjpfan@gmail.com>
Co-authored-by: Joel Nothman <joel.nothman@gmail.com>

TomDLT committed:
DOC add alfaro96 and lorentzenchr (#18178)

lopeLH committed:
FEA Add PolynomialCountSketch to Kernel Approximation module (#13003)

* Add Tensor Sketch algorithm

* Add user guide entry

* Add example

* Add benchmark

Co-authored-by: Christian Lorentzen <lorentzen.ch@googlemail.com>
Co-authored-by: Tom Dupré la Tour <tom.dupre-la-tour@m4x.org>
Co-authored-by: Roman Yurchak <rth.yurchak@gmail.com>

thomasjpfan committed:
DOC Fixes failing examples (#18180)

thomasjpfan committed:
MNT Labels pull requests based on title with regex (#18168)

sunitharavi9 committed:
DOC mention LOOCV instead of GCV in RidgeCV (#18154)

indecisiveuser committed:
DOC fix formatting docstrings in pca module (#18182)

thomasjpfan committed:
FIX Adds remainder in column transformer repr_html (#18167)

glemaitre committed:
DOC correct formatting of multilabel confusion matrix (#18193)

glemaitre committed:
ENH follow-up style improvements of DET curves #10591 (#18169)

* follow-up on #10591 

* DOC improvements

* TST add additional test for pos_label

Co-authored-by: Christian Lorentzen <lorentzen.ch@gmail.com>

alfaro96 committed:
DOC/FIX Remove toctree from autosummary directives outside of the API (#17358)

lucyleeow committed:
DOC Amend out_of_bounds docstring in IsotonicRegression (#18200)

* clarify out of bounds

* suggestions

OlehKSS committed:
DOC fixes style in develop.rst to follow documentation guideline (#18191)

Co-authored-by: Oleh Kozynets <olehkss@github.com>
Co-authored-by: Guillaume Lemaitre <g.lemaitre58@gmail.com>

coelhudo committed:
DOC fix KernelDensity docstring example (#18211)

franslarsson committed:
DOC fix docstring in utils._random.pyx following doc guideline (#18208)

alfaro96 committed:
DOC Fix documentation of the dummy module (#17306)

eschibli committed:
DOC Added explicit min_features variable to RFECV example (#18091)

Co-authored-by: Thomas J. Fan <thomasjpfan@gmail.com>

franslarsson committed:
DOC Fix docstring in sklearn.metrics.pairwise.py (#18092)

gwarmstrong committed:
ENH Add min_categories parameter for each feature to CategoricalNB (#16326)

Co-authored-by: Joel Nothman <joel.nothman@gmail.com>
Co-authored-by: Guillaume Lemaitre <g.lemaitre58@gmail.com>

thomasjpfan committed:
TST Uses regex for flaky test (#18209)

OlehKSS committed:
DOC fixes docstrings in _unsupervised module (#18214)

Co-authored-by: Oleh Kozynets <olehkss@github.com>
Co-authored-by: Juan Carlos Alfaro Jiménez <JuanCarlos.Alfaro@uclm.es>

cmarmo committed:
MNT/DOC Add preamble adding hbox around href. Fix unnumbered ref. (#18213)

NicolasHug committed:
MNT remove duplicated call to children_impurity() in tree code (#18203)

Co-authored-by: Juan Carlos Alfaro Jiménez <JuanCarlos.Alfaro@uclm.es>

franslarsson committed:
DOC Fix docstring binning.py module (#18207)

Co-authored-by: Juan Carlos Alfaro Jiménez <JuanCarlos.Alfaro@uclm.es>

sadakmed committed:
DOC fix doctstring in neighbors._base.py module (#18078)

Co-authored-by: Frans Larsson <frans.larsson@gmail.com>
Co-authored-by: Guillaume Lemaitre <g.lemaitre58@gmail.com>

sadakmed committed:
DOC fix docstring in binning.py file (#18063)

Co-authored-by: Frans Larsson <frans.larsson@gmail.com>
Co-authored-by: Guillaume Lemaitre <g.lemaitre58@gmail.com>

marenwestermann committed:
DOC shorten links in doc/* modules (#17820)

* shorten links in doc modules

* fix links

* implement suggested changes

* try to fix links with tilde sign

* change back broken links in computing.rst and try make link in faq.rst work

* implement reviewer's suggested changes

thomasjpfan committed:
MNT Only reset on the first call to partial_fit in birch (#18014)

sadakmed committed:
DOC fixed linear_model._least_angle.py parameters consistency (#18062)

Co-authored-by: Guillaume Lemaitre <g.lemaitre58@gmail.com>

NicolasHug committed:
MNT change bin_thresholds param name (#18179)

TahiriNadia committed:
DOC fix docstring in _dict_learning.py module (#17897)

Co-authored-by: Guillaume Lemaitre <g.lemaitre58@gmail.com>

NicolasHug committed:
[MRG] FIX len() of ParameterSampler (#18222)

* fix len

* whatsnew

viniciusriosfuck committed:
DOC Add legend and axis label in nearest neighbors example (#18164)

Co-authored-by: Guillaume Lemaitre <g.lemaitre58@gmail.com>

NicolasHug committed:
FEA Add sequential feature selection transformer (#17159)

Co-authored-by: rasbt <mail@sebastianraschka.com>
Co-authored-by: Thomas J. Fan <thomasjpfan@gmail.com>
Co-authored-by: Guillaume Lemaitre <g.lemaitre58@gmail.com>

franslarsson committed:
DOC Fix doc of defaults in _label.py (#18220)

Co-authored-by: Juan Carlos Alfaro Jiménez <JuanCarlos.Alfaro@uclm.es>
Co-authored-by: Guillaume Lemaitre <g.lemaitre58@gmail.com>

crypdick committed:
ENH added zero_division param to jaccard metric (#17866)

Co-authored-by: Joseph Willard <joseph@sirbotsy.sirbotsy>
Co-authored-by: Thomas J. Fan <thomasjpfan@gmail.com>
Co-authored-by: Guillaume Lemaitre <g.lemaitre58@gmail.com>

alfaro96 committed:
DOC Fix documentation of the base module (#17548)

Co-authored-by: Guillaume Lemaitre <g.lemaitre58@gmail.com>

clmbst committed:
DOC Fix documentation of default values in decomposition module (#17739)

Co-authored-by: Jérémie du Boisberranger <34657725+jeremiedbb@users.noreply.github.com>
Co-authored-by: Guillaume Lemaitre <g.lemaitre58@gmail.com>

franslarsson committed:
DOC fix docstring in covariance module (#18227)

franslarsson committed:
DOC fix docstring in ensemble module (#18228)

franslarsson committed:
DOC Add None as valid option to random_state (#18234)

franslarsson committed:
DOC Add None as valid option to random_state in svm module (#18233)

franslarsson committed:
DOC add None as valid option to random_state in model_selection module (#18232)

OlehKSS committed:
DOC Fix docs for clustering.rst following doc guideline (#18230)

Co-authored-by: Oleh Kozynets <olehkss@github.com>

franslarsson committed:
DOC Add None as valid option to random_state for metrics module (#18236)

franslarsson committed:
DOC Add None as valid option to random_state for manifold module (#18237)

franslarsson committed:
DOC fix docstring in preprocessing module following doc guideline (#18239)

franslarsson committed:
DOC Add None as valid option to random_state for linear_model module (#18240)

franslarsson committed:
DOC fix docstring for datasets module following doc guideline (#18241)

Ultramann committed:
FIX fix hinge_loss when labels are provided and not inferred from y_true (#17935)

Co-authored-by: Ultramann <cgoltermann@kpmg.com>
Co-authored-by: Guillaume Lemaitre <g.lemaitre58@gmail.com>

franslarsson committed:
DOC Add None as valid option to random_state in neural_network module (#18246)

franslarsson committed:
DOC Add None as valid option to random_state in neighbors module (#18247)

cmarmo commented:
Hi @okz12 , sorry for the late answer and thanks for your patience!
If you are still interested in working on this, do you mind fixing conflicts? Then we can check why tests are failing. Thanks!

okz12 commented:
Hi @cmarmo,
Sure, I'll take a look and fix the conflicts - need to double check the changes since it's been a while.

franslarsson committed:
DOC Add None as valid option to random_state for dummy.py (#18245)

alfaro96 committed:
MNT Label pull requests with CI regex title (#18249)

franslarsson committed:
DOC Add None as valid option to random_state for utils module (#18253)

avinash-ak committed:
DOC minor improvements in  CODE_OF_CONDUCT.md file (#18188)

cool-RR committed:
MNT Use raise from in 19 modules (#17835)

franslarsson committed:
DOC Add None as valid option to random_state in random_projection.py (#18238)

Co-authored-by: Guillaume Lemaitre <g.lemaitre58@gmail.com>

Ultramann committed:
EHN raise error with inconsistant transformer_weight in FeatureUnion (#17876)

Co-authored-by: Cary Goltermann <cgoltermann@kpmg.com>
Co-authored-by: Guillaume Lemaitre <g.lemaitre58@gmail.com>
Co-authored-by: Thomas J. Fan <thomasjpfan@gmail.com>

david-cortes committed:
ENH Do not pass random_state to estimator in IterativeImputer (#15636)

alexshacked committed:
FIX Raise proper error for sparse dataframe with mixed dtypes (#17992)

Co-authored-by: Guillaume Lemaitre <g.lemaitre58@gmail.com>
Co-authored-by: Thomas J. Fan <thomasjpfan@gmail.com>

OlehKSS committed:
DOC Fix docstring in examples (#18251)

Co-authored-by: Oleh Kozynets <olehkss@github.com>

franslarsson committed:
DOC Add None as valid option to random_state for gaussian_process module (#18252)

franslarsson committed:
DOC Fix docstring in grower.py following doc guideline (#18206)

Co-authored-by: Juan Carlos Alfaro Jiménez <JuanCarlos.Alfaro@uclm.es>
Co-authored-by: Guillaume Lemaitre <g.lemaitre58@gmail.com>

venkyyuvy committed:
EHN make some meta-estimators lenient towards missing values (#17987)

Co-authored-by: Guillaume Lemaitre <g.lemaitre58@gmail.com>

smarie committed:
TST Fixed issue with small_pos_ratio on mac os x with python >= 3.7 due to floating point precision issue (#18270)

Co-authored-by: Sylvain MARIE <sylvain.marie@se.com>

chaitanyamogal committed:
DOC fix docsting of fit_transform in Voting estimator (#18185)

marenwestermann committed:
DOC shorten links in plot_partial_dependence.py (#18273)

alexshacked committed:
FIX be more permissive with using subtype in test_check_pandas_sparse_valid (#18277)

lucyleeow committed:
DOC Clarify shape in OVO and OVR decision function docstring (#18279)

thomasjpfan committed:
MNT Update license formating for github to recognize it (#18283)

thomasjpfan committed:
FIX Correctly sets MDS pairwise attribute (#18278)

glemaitre committed:
ENH Adds plot_det_curve and associated display (#18176)

Co-authored-by: Christian Lorentzen <lorentzen.ch@gmail.com>
Co-authored-by: Thomas J. Fan <thomasjpfan@gmail.com>

cmarmo committed:
DOC Tutorial structure and layout (#18261)

Co-authored-by: Thomas J. Fan <thomasjpfan@gmail.com>

subrat93 committed:
ENH Raises warning when getting non-finite score in SearchCV (#18266)

Co-authored-by: Guillaume Lemaitre <g.lemaitre58@gmail.com>
Co-authored-by: Thomas J. Fan <thomasjpfan@gmail.com>

franslarsson committed:
DOC Add None as option for random_state in affinity_propagation (#18306)

marenwestermann committed:
DOC alphas_ documentation of LassoLarsIC (#18224)

anandt555 committed:
DOC fix erroneous statement regarding meta-estimator in all_estimators (#18231)

Co-authored-by: anandt <anandt@cvent.com>
Co-authored-by: Guillaume Lemaitre <g.lemaitre58@gmail.com>

albertvillanova committed:
DOC fix hyperlink in whats new 0.19 (#18303)

okz12 committed:
Merged feature branch with new master

okz12 committed:
Merge with master

okz12 commented:
I tried to update the branch/merge by rebasing to the current master but it seems to have brought in all the other commits in the last 8 months. I'll try branching from the current master and adding the changes there and post the new PR here.
